nms/app/packages/magmalte/app/components/context/LteNetworkContext.js

```diff
@@ -14,14 +14,17 @@
  * @format
  */
 'use strict';
-import type {UpdateNetworkProps} from '../../state/lte/NetworkState';
-import type {lte_network} from '@fbcnms/magma-api';
+import type {UpdateNetworkProps as FegLteUpdateNetworkProps} from '../../state/feg_lte/NetworkState';
+import type {UpdateNetworkProps as LteUpdateNetworkProps} from '../../state/lte/NetworkState';
+import type {feg_lte_network, lte_network} from '@fbcnms/magma-api';
 
 import React from 'react';
 
 export type LteNetworkContextType = {
-  state: lte_network,
-  updateNetworks: (props: $Shape<UpdateNetworkProps>) => Promise<void>,
+  state: $Shape<lte_network & feg_lte_network>,
```

Comment:
This type is used multiple times in multiple places. You should pull it out to be its own proper type. Also I'm not entirely sure what `$Shape` does, but if you just want to union types, you can do `type TypeC = TypeA & TypeB`

Reply:
Shape allows little bit of a loose definition of the type, i.e we can have few missing attributes. For e.g. lte_networks contain "subscriber_config" attribute and feg_lte networks contain federation_config attribute. We want to allow combination of both but not all attributes. Hence came up with this. I can typedef $Shape<lte_network & feg_lte_network> something meaningful. If I have to accommodate the flow definitions properly then this complicates the code flow and I have to create React components with appropriate flow definitions everywhere. Do you have other suggestions on how I can fix this or how such pattern is typically handled ?